x/subscription/README.md

```diff
@@ -0,0 +1,202 @@
+
+3. **Handle Special Cases for Subscriptions:**
+
+   - If a future subscription is set:
+     - Activate the future subscription by updating current subscription details with those of the future subscription and restart the subscription.
+   - Else, if auto-renewal is enabled:
+     - Attempt to renew the subscription for another month.
+     - If renewal fails, remove the expired subscription.
+   - If there is no future subscription or auto-renewal, remove the expired subscription.
+
+### Subscription Upgrade
+
+A subscription can be upgraded to a more expensive plan.
+
+That also means, that if a consumer buys a plan, and then decides to downgrade their plan, the way to do it is either buy more months in the cheaper subscription, so the amount that the consumer will pay will be larger than the one using the same command as shown above, with the only caveat - the `plan-index` must be different than the currently active subscription's plan, and, it must be with a higher price.
+
+Tokens are deducted from the creator's account immediately, and the new plan becomes effective in the next epoch.  
+The old subscription will still be saved in the fixation store, for buffering purposes (the amount of blocks that the subscription will be saved for, is determined in the `BlocksToSave` function, under the [EpochStorage](https://github.com/lavanet/lava/blob/main/x/epochstorage/README.md) module).
+
+More ways to upgrade are by making an [Advance Purchase](#advance-purchase) or enabling [Auto Renewal](#auto-renewal).
+
+### Subscription Renewal
+
+Users have the option to renew their subscription either manually or automatically. To renew manually, users can utilize the same command as demonstrated above. They simply need to adjust the 'plan-index' to match the plan of their currently active subscription. Additionally, they can set the duration, allowing the new duration to be added to the remaining duration of the active subscription.
+To renew automatically, please refer to [Auto Renewal](#auto-renewal).
+
+### Advance Purchase
+
+Users can purchase several months of subscription in advance, using the subscription buy transaction command like so:
+
+```bash
+lavad tx subscription buy --advance-purchase [plan-index] [optional: consumer] [optional: duration(months)] [flags]
+```
+
+This will create the `FutureSubscription` object inside the `Subscription` object, as can be seen above.
+The new subscription will be triggered once the current active subscription expires.
+
+Users can decide on the duration of this future subscription, and they'll pay the full amount for the entire period immediately.
+If the plan changes during that period, users who bought this plan before it changed using the `--advance-purchase` flag, they won't be affected by that change.
+
+If a user tries to replace the future subscription with another, the new plan's price must be higher, considering the amount of days bought.
```

Comment:
bough -> bought

Reply:
I think this is already bought, isn't it?